src/internal/cache-client.ts

```diff
@@ -208,36 +213,37 @@ export class CacheClient {
         },
         (err, resp) => {
           if (resp) {
-            const momentoResult = momentoResultConverter(resp.result);
-            if (
-              momentoResult !== CacheGetStatus.Miss &&
-              momentoResult !== CacheGetStatus.Hit
-            ) {
-              reject(new UnknownServiceError(resp.message));
+            switch (resp.result) {
+              case cache.cache_client.ECacheResult.Miss:
+                resolve(new CacheGet.Miss());
+                break;
+              case cache.cache_client.ECacheResult.Hit:
+                resolve(new CacheGet.Hit(resp.cache_body));
+                break;
+              case cache.cache_client.ECacheResult.Invalid:
+                resolve(new CacheGet.Error(new UnknownError(resp.message)));
+                break;
+              case cache.cache_client.ECacheResult.Ok:
+                resolve(new CacheGet.Error(new UnknownError(resp.message)));
+                break;
```

Comment:
The two cases seem the same. Use a fall through?
```suggestion
              case cache.cache_client.ECacheResult.Invalid:
              case cache.cache_client.ECacheResult.Ok:
                resolve(new CacheGet.Error(new UnknownError(resp.message)));
                break;
```

Reply:
Sure, will do.